stepper-2.0.md

```diff
@@ -59,42 +59,58 @@ Protocol
 ```
 0  START_SYSEX                             (0xF0)
 1  Stepper Command                         (0x62)
-2  stop command                            (0x01)
+2  zero command                            (0x01)
 3  device number                           (0-9)
 4  END_SYSEX                               (0xF7)
 ```
 
-**Stepper step**
+**Stepper step (relative move)**
+
+Position is specified as a 32-bit unsigned long. Speed is specified as a
+16-bit unsigned int.
+
 ```
 0  START_SYSEX                             (0xF0)
 1  Stepper Command                         (0x62)
 2  step command                            (0x02)
 3  device number                           (0-9)
 4  direction                               (0-1) (0x00 = CW, 0x01 = CCW)
-5  num steps byte1 LSB
-6  num steps byte2
-7  num steps byte3 MSB                     (21 bits (2,097,151 steps max))
-8  num speed LSB                           (steps per second)
-9  num speed MSB
-10 END_SYSEX                               (0xF7)
+5  num steps, bits 0-6
+6  num steps, bits 7-13
+7  num steps, bits 14-20
+8  num steps, bits 21-27
+9  num steps, bits 28-32
+10 speed, bits 0-6                         (steps per second * 1000)
```

Comment:
Okay, I've given it a lot of thought and here's where I landed. I wanted to pack as much as I could into a multiple of 7 but still handle the most unlikely of extremes a user might need.

The only things we need a float for are acceleration and speed. The ranges we might need for speed are much greater than what would ever need for acceleration. The max speed value accelStepper can handle on an Arduino is about 1000 steps per second so we can bias our range to the low side of things.

With a 16 bit mantissa we would get just five digits of precision. That means if we wanted something to move one step per day we would be a half second off every day, or about 3 minutes per year (assuming a perfect clock on the Arduino). Therefore I think we should have a 23 bit mantissa. That gives us the full precision of a IEEE 754 integer.

If we then have a 4 bit exponent value (biased -18, with an implicit 1's bit) and a sign bit we get a healthy range of +/- 83886.08 with minimum values that are so small we will never really need to worry about them. This all packs perfectly into 4 7-bit bytes.

Acceleration is trickier since the achieved speed will be dependent on the duration of the move, however I am confident that the same approach with a 16 bit mantissa and a 4 bit exponent (biased -16 with an implicit 1's bit) would give us all the range we ever need while still packing nicely into 3 bytes.